File: Andromeda.AvaloniaApp.FSharp/Converter/ProductInfoToStringConverter.cs
using Avalonia;
using Avalonia.Controls;
using Avalonia.Data.Converters;
using Avalonia.Media.Imaging;
using Microsoft.FSharp.Core;
using System;
using System.Globalization;

namespace Andromeda.AvaloniaApp.Converter
{
    public class ProductInfoToStringConverter : IValueConverter
    {
        public object Convert(object value, Type targetType,
            object parameter, CultureInfo culture)
        {
            if (value != null) {
                var info = (GogApi.DotNet.FSharp.Listing.ProductInfo) value;
                return info.title;
            } else {
                return null;
            }
        }

        public object ConvertBack(object value, Type targetType,
            object parameter, CultureInfo culture)
        {
            return null;
        }
    }
}

File: Andromeda.AvaloniaApp.FSharp/Converter/ProductInfoToStringConverter.fs
namespace Andromeda.AvaloniaApp.FSharp.Converter

open Avalonia.Data.Converters
open GogApi.DotNet.FSharp.Listing
open Microsoft.FSharp.Core

type ProductInfoToStringConverter () =
    interface IValueConverter with
        member __.Convert (value, _, _, _) =
            match value with
            | null -> null
            | :? ProductInfo as value -> value.title :> obj
            | _ -> invalidArg "value" ""

        member __.ConvertBack (_, _, _, _) =
            invalidOp ""
